source/features/comment-to-draft-pr-button.tsx

```diff
@@ -0,0 +1,45 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import onNewComments from '../github-events/on-new-comments';
+
+function handleClicks(): void {
+	const buttonsSingleComment = select.all([
+		'table:not([style="display: none"]) .review-simple-reply-button',
+		'table:not([style="display: none"]) .add-comment-label' // For when adding a single comment after having started a review
+	]);
+	for (const element of buttonsSingleComment) {
+		element.innerHTML = ' Add single comment to Draft PR ';
+	}
+
+	const buttonsStartReview = select.all('table:not([style="display: none"]) .start-review-label');
+	for (const element of buttonsStartReview) {
+		element.innerHTML = ' Start a review to Draft PR ';
+	}
+}
+
+function init(): void {
+	let commentButton;
+	if (pageDetect.isPRConversation()) {
+		commentButton = select('#partial-new-comment-form-actions .btn-primary');
+		commentButton!.innerHTML = 'Comment to Draft PR';
+	} else if (pageDetect.isPRFiles()) {
+		delegate('.diff-table', 'button', 'click', handleClicks);
```

Comment:
I don't think that's the right way. We add custom buttons like `table-input` on the comment forms on the Files tab. You can copy what they use

Reply:
Sorry, but I don't quite understand.

With this feature, the purpose is to simply change the name of the buttons, 
not (remove the existing ones and) add new buttons with a different name.
Are you suggesting to use `remove()` and `.after()`? 
https://github.com/sindresorhus/refined-github/blob/ae07d212f364ba0193958b95729e4e99d4653b90/source/features/table-input.tsx#L40-L68

Reply:
It looks like a plain loop is all that should be needed and GitHub will preserve the changes, no need for onNewComments at all:

https://github.com/sindresorhus/refined-github/blob/ae07d212f364ba0193958b95729e4e99d4653b90/source/features/table-input.tsx#L39-L40

Reply:
Regarding `onNewComments`:  I tested removing it, but then, in 'Conversations' tab, the 'Comment' button name no longer change after submitting a new comment. Therefore it seems necessary.

And regarding my use of `delegate`:
in 'Files 'tab, e.g. https://github.com/darkred/test/pull/6/files
initially, there's no comment textarea,
and no `table:not([style="display: none"]) .review-simple-reply-button/.start-review-label` button elements:
these elements only exist *after* you click in a row ([capture](https://user-images.githubusercontent.com/723651/123557151-fc6d7f00-d797-11eb-89e1-03b0a892464b.gif))
therefore I think it's necessary to delegate clicks on the .diff-table element.
You can't loop on them, before they exist.

Am I not right? Have I got something wrong?

Reply:
Where in `table-input` is there `:not([style="display: none"])`? It's not there because it's not needed. As I tried saying before, there's a template that GitHub copies, including all the changes we make to it. Your selector explicitly excludes it.

Same goes for `onNewComments` in this feature, it's not needed.

Reply:
Maybe it needs dontDeduplicate. (I forgot what we called it)

Reply:
For the 'Files' tab, `onNewComments` is indeed not needed::
I had used the `:not([style="display: none"])` selector part
because I wrongly thought that these hidden elements shouldn't also match my selector.
It indeed works ok after removing that selector part + `onNewComments`.
I just removed the selector part.

BUT, on the Conversations tab, I still believe the `onNewComments` is still needed:
My relevant selector is: `#partial-new-comment-form-actions .btn-primary`. 
There's no `:not([style="display: none"])` in it, and it only matches 1 element, the Comment textarea to the bottom.
I can't find any hidden element that is a "template comment box element" 
that can also be selected with a variation of my selector above.
Therefore I can't seem to make use of a `for (const el of __)` loop.

Reply:
Also, I tried adding the following inside `void features.add(` but it didn't help:

```js
	deduplicate: 'false',
```

Reply:
Regarding the Conversations tab, maybe this hidden 'template comment box element' contains code only for the editor's toolbar buttons (a button of which, `md-task-list`, is used as reference in [table-input](https://github.com/sindresorhus/refined-github/blob/main/source/features/table-input.tsx)), 
but not for the 'Comment' button (or the 'Close pull request' one) too?

I couldn't find any such hidden additional button, as I describe in my previous comment.